web/l10n.js

```diff
@@ -0,0 +1,95 @@
+/* Copyright 2023 Mozilla Foundation
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/** @typedef {import("./interfaces").IL10n} IL10n */
+/**
+ * @implements {IL10n}
+ */
+class L10n {
+  #dir;
+
+  #l10n;
+
+  constructor({ lang, isRTL }, l10n = null) {
+    this.lang = L10n.#fixupLangCode(lang);
```

Comment:
Given that this property ideally should be private as well (but cannot be that given the `GenericL10n` constructor), can we at least make it "semi"-private?
```suggestion
    this._lang = L10n.#fixupLangCode(lang);
```